qutebrowser/mainwindow/mainwindow.py

```diff
@@ -472,6 +472,17 @@ def showEvent(self, e):
         super().showEvent(e)
         objreg.register('last-visible-main-window', self, update=True)
 
+    def changeEvent(self, e):
+        """Extend changeEvent to register an activationChange.
+
+        Args:
+            e: The QChangeEvent
+        """
+        super().changeEvent(e)
+        if e.type() == QEvent.ActivationChange and self.isActiveWindow():
+            log.misc.debug("Window activation changed to {!r} (win_id={})"
+                           .format(self, self.win_id))
+
```

Comment:
I think I tried that at first, but it didn't seem to work reliably. Either there is a difference in the order in which those events are processed (so we signal that activation has changed, but `window.isActivated` still returns false), or some other things aren't right. I figured just using the right event would probably be easier.